CrazyCanvas/Source/ECS/Systems/Player/WeaponSystem.cpp

```diff
@@ -115,6 +120,10 @@ void WeaponSystem::Fire(WeaponComponent& weaponComponent, const glm::vec3& start
 	ECSCore* pECS = ECSCore::GetInstance();
 	const Entity projectileEntity = pECS->CreateEntity();
 
+	// Get the firing player's team index
+	const uint32 playerTeam = pECS->GetConstComponent<TeamComponent>(weaponComponent.WeaponOwner).TeamIndex;
```

Comment:
How many TeamIndex's do we need? I think uint8 would be enough for handling different teams